lib/Controller/APIController.php

```diff
@@ -71,15 +82,35 @@ public function getRetentions(): DataResponse {
 		return new DataResponse($result);
 	}
 
-	public function addRetention(int $tagid, int $timeunit, int $timeamount, int $timeafter = Constants::CTIME): DataResponse {
+	/**
+	 * Create a retention rule
+	 *
+	 * @param int $tagid Tag the retention is based on
+	 * @param int<0, 3> $timeunit Time unit of the retention (days, weeks, months, years)
+	 * @psalm-param Constants::UNIT_* $timeunit
+	 * @param positive-int $timeamount Amount of time units that have to be passed
+	 * @param int<0, 1> $timeafter Whether retention time is based creation time (0) or modification time (1)
```

Comment:
```suggestion
	 * @param 0|1|2|3 $timeunit Time unit of the retention (days, weeks, months, years)
	 * @psalm-param Constants::UNIT_* $timeunit
	 * @param positive-int $timeamount Amount of time units that have to be passed
	 * @param 0|1 $timeafter Whether retention time is based creation time (0) or modification time (1)
```
I think distinct values are better, because each value has a specific meaning (so enum type) and it's not just minimum/maximum that is important.